app/Http/Controllers/Api/UserController.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\Api\UserRequest;
+use App\Http\Requests\Api\UserRoleRequest;
+use App\Models\User;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Hash;
+
+class UserController extends Controller
+{
+    public function index(): Response
+    {
+        return response(User::orderBy('name')->paginate(100));
+    }
+
+    public function update(UserRequest $request, User $user): Response
+    {
+        $updatedPassword = Hash::make($request->password);
```

Comment:
what's the difference between `$validated` object & `$request` object?

Reply:
`$request` object represent the entire incoming request information without validation rules which is defined in the `UserRequest` class. and the `$validated` object specifically represents the data that has passed the validation rules which is defined in the `UserRequest` class.